Initial draft to calculate node latency (#687)
kube-burner/kube-burner: docs/measurements.md

In case of not meeting any of the configured thresholds, like the example above, **kube-burner return code will be 1**.

> ## Node latency

Reviewer: Can we rename this feature to something like, for example `nodeConditionsTimings`?, I think that node latency ca be confusing, as a k8s node is not an entity that evolves through multiple life cycles like a pod but an entity that can be under different conditions.

Author: Renaming it to `nodeConditionTimings` might be confusing because we are not reporting the timestamps instead we are reporting the latencies. Though nodes do not follow a particular order during their lifecycle like pods, they only reaches to its `Ready` state only when their pre-requisite events are completed. So I think its good to keep the name as is.

> Collects latencies from the different node startup phases, these **latency metrics are in ms**. It can be enabled with:

Reviewer: ```suggestion
Collects timing information about the different node conditions of the cluster,  **collected metrics are in ms**. It can be enabled with:
```

```yaml
  measurements:
  - name: nodeLatency
```

### Metrics

> The metrics collected are pod latency timeseries (`nodeLatencyMeasurement`) and four documents holding a summary with different pod latency quantiles of each pod condition (`nodeLatencyQuantilesMeasurement`).

Reviewer: ```suggestion
The metrics collected are node latency timeseries (`nodeLatencyMeasurement`) and four documents holding a summary with different latency quantiles of each node condition (`nodeLatencyQuantilesMeasurement`).
```

```json
{
  "timestamp": "2024-08-25T12:49:26Z",
  "nodeMemoryPressureLatency": 0,
  "nodeDiskPressureLatency": 0,
  "nodePIDPressureLatency": 0,
  "nodeReadyLatency": 82000,
  "metricName": "nodeLatencyMeasurement",
  "uuid": "4f9e462c-cacc-4695-95db-adfb841e0980",
  "jobName": "namespaced",
  "nodeName": "ip-10-0-34-104.us-west-2.compute.internal",
  "labels": {
    "beta.kubernetes.io/arch": "amd64",
    "beta.kubernetes.io/instance-type": "m6i.xlarge",
    "beta.kubernetes.io/os": "linux",
    "failure-domain.beta.kubernetes.io/region": "us-west-2",
    "failure-domain.beta.kubernetes.io/zone": "us-west-2b",
    "kubernetes.io/arch": "amd64",
    "kubernetes.io/hostname": "ip-10-0-34-104.us-west-2.compute.internal",
    "kubernetes.io/os": "linux",
    "node-role.kubernetes.io/worker": "",
    "node.kubernetes.io/instance-type": "m6i.xlarge",
    "node.openshift.io/os_id": "rhcos",
    "topology.ebs.csi.aws.com/zone": "us-west-2b",
    "topology.kubernetes.io/region": "us-west-2",
    "topology.kubernetes.io/zone": "us-west-2b"
  }
}
```

---

> Pod latency quantile sample:

Reviewer: ```suggestion
Node latency quantile sample:
```

```json
{
  "quantileName": "Ready",
  "uuid": "4f9e462c-cacc-4695-95db-adfb841e0980",
  "P99": 163000,
  "P95": 163000,
  "P50": 93000,
  "max": 163000,
  "avg": 122500,
  "timestamp": "2024-08-25T20:42:59.422208263Z",
  "metricName": "nodeLatencyQuantilesMeasurement",
  "jobName": "namespaced",
  "metadata": {}
},
{
  "quantileName": "MemoryPressure",
  "uuid": "4f9e462c-cacc-4695-95db-adfb841e0980",
  "P99": 0,
  "P95": 0,
  "P50": 0,
  "max": 0,
  "avg": 0,
  "timestamp": "2024-08-25T20:42:59.422209628Z",
  "metricName": "nodeLatencyQuantilesMeasurement",
  "jobName": "namespaced",
  "metadata": {}
}
```

> Where `quantileName` matches with the node conditions and can be:

Reviewer: I think it can be a good practice to add a link to the official docs to explain node conditions https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#node-conditions
